Tests/TuistKitTests/Services/GraphServiceTests.swift

```diff
@@ -64,4 +64,33 @@ final class GraphServiceTests: TuistUnitTestCase {
         Graph exported to \(graphPath.pathString)
         """)
     }
+    
+    func test_run_whenJson() throws {
+        // Given
+        let temporaryPath = try self.temporaryPath()
+        let graphPath = temporaryPath.appending(component: "graph.json")
+        let projectManifestPath = temporaryPath.appending(component: "Project.swift")
+
+        try FileHandler.shared.touch(graphPath)
+        try FileHandler.shared.touch(projectManifestPath)
+
+        // When
+        try subject.run(
+            format: .json,
+            layoutAlgorithm: .dot,
+            skipTestTargets: false,
+            skipExternalDependencies: false,
+            targetsToFilter: [],
+            path: temporaryPath,
+            outputPath: temporaryPath
+        )
+        let got = try FileHandler.shared.readTextFile(graphPath)
+        let expected = "{\"packages\":[],\"workspace\":{\"path\":\"\\/\",\"schemes\":[],\"additionalFiles\":[],\"name\":\"test\",\"projects\":[],\"xcWorkspacePath\":\"\\/\"},\"targets\":[],\"path\":\"\\/\",\"dependencies\":[],\"name\":\"graph\",\"projects\":[]}"
```

Comment:
Comparing the raw JSON string will make the test cumbersome to maintain because changes in the graph will lead to changes in the string and thus failing tests. What about serializing that string back into a graph model?

Reply:
Yeah sure. I thought comparing the direct output would be more accurate. But I suppose deserialize it again would work too.